### PR TITLE
add some test case

### PR DESCRIPTION
**Pull request by xiaoge1001**
As shown in the title, I added a test case.

### CODE REVIEW

**Comment by xiaoge1001** [comment]
How should I know what went wrong?

图片

https://readthedocs.com/projects/canonical-cloud-init/builds/1220059/
图片
1

File: tests/unittests/test_atomic_helper.py [hunk 1]
```diff
@@ -30,6 +30,16 @@ def test_file_permissions(self):
         atomic_helper.write_file(path, contents, mode=0o400)
         self.check_file(path, contents, perms=0o400)
 
+    def test_file_preserve_permissions(self):
+        """create a file with mode 700, then write_file with mode 644."""
+        path = self.tmp_path("test_file_preserve_permissions")
+        contents = b"test_file_perms"
+        f = open(path, mode="wb")
```
[the remark this answers is not in the document]
Comment (xiaoge1001) [reply tests/unittests/test_atomic_helper.py:37]: Ok, thank you very much. I will modify it

**Comment by xiaoge1001** [comment]
https://github.com/canonical/cloud-init/actions/runs/3696086011/jobs/6259314258
图片

https://app.travis-ci.com/github/canonical/cloud-init/jobs/591191519
图片

The failure of CI should have nothing to do with my PR, how trigger again?

**Comment by xiaoge1001** [comment]
I add test for util.human2bytes() function.

**Comment by TheRealFalcon** [comment]
To re-trigger CI, we usually close the PR and then immediately re-open it.

**Comment by holmanb** [comment]
> The failure of CI should have nothing to do with my PR

Agreed :/ We get more ephemeral failures related to snapcraft and launchpad than I would like.

> how trigger again?

The easiest way to retrigger is either of:

1. add commit (force push/squash/no-op commit, or if you have more to add)
2. close/reopen the PR

Not the best UI, but it works.

As for the docs thing you mentioned earlier, that's been identified - stay tuned.

**Comment by xiaoge1001** [comment]
All inspection items pass

**Review by holmanb — APPROVED** [review]
Thanks for the contributions @xiaoge1001!

This looks good to me.

**Pull request merged**